cpp/src/copying/concatenate.cu

```diff
@@ -354,13 +354,19 @@ concatenate(std::vector<column_view> const& columns_to_concat,
             rmm::mr::device_memory_resource *mr,
             cudaStream_t stream) {
 
-  if (columns_to_concat.empty()) { return std::make_unique<column>(); }
+  CUDF_EXPECTS(not columns_to_concat.empty(),
+               "Unexpected empty list of columns to concatenate.");
```

Comment:
If there's no columns to concat can we not just return an empty table? I could imagine from the python side:
```
cudf.concat([cudf.DataFrame(), cudf.DataFrame()])
```

Reply:
This function returns a column. If there aren't any columns to concat, we don't know the type of the column to return.

Reply:
If you try and concat zero tables you'll get a table with zero columns in return: https://github.com/rapidsai/cudf/blob/2f136f0893bee595853e3080afe0619e2e4bfc09/cpp/src/copying/concatenate.cu#L422

Reply:
Woops. Is this situation handled at a higher level in the `Table` concatenate API?

Reply:
Yes, see above.